app/services/add_ephemera_to_collection.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+class AddEphemeraToCollection
+  attr_reader :project_id, :collection_id, :change_set_persister, :logger, :query_service
+  def initialize(project_id:, collection_id:, change_set_persister:, logger: Valkyrie.logger)
+    @project_id = project_id
+    @collection_id = collection_id
+    @change_set_persister = change_set_persister
+    @logger = logger
+    @query_service = Valkyrie.config.metadata_adapter.query_service
+  end
+
+  def process_box(box)
+    logger.info("Processing a box: #{box.decorate.title}")
+    box.decorate.members.each do |folder|
+      process_folder(folder)
+    end
+  end
+
+  def process_folder(folder)
+    logger.info("Processing a folder: #{folder.decorate.title.first}")
+    col_ids = folder.member_of_collection_ids + [collection_id]
+    change_set = ChangeSet.for(folder)
+    change_set_persister.save(change_set: change_set) if change_set.validate(member_of_collection_ids: col_ids)
+  end
+
+  def perform_now
+    project = query_service.find_by(id: project_id)
+    collection = query_service.find_by(id: collection_id)
+    logger.info("Adding members of Ephemera Project #{project.decorate.title} to Collection #{collection.decorate.title}")
+    project.decorate.members.each do |member|
```

Comment:
I think you should use the `EphemeraProjectDecorator.folders` and `EphemeraProjectDecorator.boxes` methods  instead of doing the `case` logic.

Reply:
Yes -- the case logic had a strong code smell.